lib/galaxy/model/__init__.py

```diff
@@ -4721,21 +4721,20 @@ def dataset_instances(self):
         else:
             return [element_object]
 
-    def copy_to_collection(self, collection, destination=None, element_destination=None, copy_dataset_instance_attributes=None, flush=True):
+    def copy_to_collection(self, collection, destination=None, element_destination=None, dataset_instance_attributes=None, flush=True):
         element_object = self.element_object
         if element_destination:
             if self.is_collection:
                 element_object = element_object.copy(
                     destination=destination,
                     element_destination=element_destination,
-                    copy_dataset_instance_attributes=copy_dataset_instance_attributes,
+                    dataset_instance_attributes=dataset_instance_attributes,
                     flush=flush
                 )
             else:
                 new_element_object = element_object.copy(flush=flush, copy_tags=element_object.tags)
-                if copy_dataset_instance_attributes is not None:
-                    if 'dbkey' in copy_dataset_instance_attributes:
-                        new_element_object.dbkey = copy_dataset_instance_attributes['dbkey']
+                for attribute, value in dataset_instance_attributes.items():
```

Comment:
I think we need some sort of validation at this stage (or earlier). Can you create a list of valid attributes that are just dbkey and throw a RequestParameterInvalidException attribute is not in this list.

Reply:
Oh never mind, you moved that validation up to the API layer. This is fine then I think.

Reply:
I was hoping we're going to move this into pydantic models down the road and deal with this in the manager.